src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random r = new Random();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure[] figures = {new Circle(colorSupplier.getRandomColor(), r.nextInt(0, 1000)),
+                new Square(colorSupplier.getRandomColor(), r.nextInt(0, 1000)),
+                new RightTriangle(colorSupplier.getRandomColor(),
+                        r.nextInt(0, 1000), r.nextInt(0, 1000)),
+                new Rectangle(colorSupplier.getRandomColor(), r.nextInt(0, 1000),
+                        r.nextInt(0, 1000)),
+                new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                    r.nextInt(0, 1000), r.nextInt(0, 1000), r.nextInt(0, 1000))};
+        return figures[r.nextInt(0, 5)];
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("white", 10);
```

Comment:
1. For magic numbers use constants, and for color you can use your enum
2. in constructor you create random color and value, you can create own methods for this and reuse it

Reply:
what is it "magic numbers"?

Reply:
what is it "magic numbers?